### PR TITLE
feat: add hook for karbon client

### PR DESCRIPTION
**Pull request by DanSnow**
## Proposal
Support hooking Karbon's requests to enable other functionalities

## Tophat
1. Run `yarn dev` in the playground and ensure that the console logs relevant requests
2. Execute `NITRO_PRESET=cloudflare-page yarn build` to confirm successful build in playground

### CODE REVIEW

File: packages/karbon/src/runtime/plugins/client-hooks.ts [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import { _karbonClientHooks } from '../composables/storipress-base-client'
+import { defineNuxtPlugin } from '#imports'
+
+export default defineNuxtPlugin((nuxt) => {
+  _karbonClientHooks.hook('karbon:request', async (ctx) => {
```
Comment (DanSnow) [inline packages/karbon/src/runtime/plugins/client-hooks.ts:5]: forward hook call to Nuxt hook

File: packages/karbon/src/runtime/server/plugins/hook.ts [hunk 2]
```diff
@@ -0,0 +1,14 @@
+import { clientHooks } from '@storipress/karbon/internal'
+import { type _karbonClientHooks } from '../../composables/storipress-base-client'
+import { defineNitroPlugin } from '#imports'
+
+const hooks: typeof _karbonClientHooks = clientHooks
+
+export default defineNitroPlugin((nitro) => {
+  hooks.hook('karbon:request', (ctx) => {
```
Comment (DanSnow) [inline packages/karbon/src/runtime/server/plugins/hook.ts:8]: forward hook call to Nitro hook

File: packages/karbon/src/runtime/composables/storipress-base-client.ts [hunk 3]
```diff
@@ -37,7 +53,75 @@ export function createTenantURL(config: Pick<ModuleRuntimeConfig['storipress'],
...
+
+    operation.setContext({
+      karbonTracing: id,
+    })
+
+    return new Observable((observer) => {
+      let subscription: Subscription
+      let closed = false
+      Promise.resolve(
+        _karbonClientHooks.callHookParallel('karbon:request', {
```
Comment (DanSnow) [inline packages/karbon/src/runtime/composables/storipress-base-client.ts:76]: hook request

File: packages/karbon/src/runtime/composables/storipress-base-client.ts [hunk 4]
```diff
@@ -37,7 +53,75 @@ export function createTenantURL(config: Pick<ModuleRuntimeConfig['storipress'],
...
+      )
+        .then(() => {
+          if (closed) {
+            return
+          }
+
+          function createCallback(type: 'next' | 'error' | 'complete', callback: (...args: any[]) => void) {
+            return (...args: any[]) => {
+              Promise.resolve(
+                _karbonClientHooks.callHookParallel('karbon:response', {
```
Comment (DanSnow) [inline packages/karbon/src/runtime/composables/storipress-base-client.ts:90]: hook response

**Review by SidStraw — APPROVED** [review]
1. Tophat pass
2. code review pass

**Pull request merged**